File: solutions/typescript/collatz-conjecture/1/collatz-conjecture.ts
export function steps(count: number): number {
  let step_count = 0;

  if ( count <= 0 ) throw('Only positive numbers are allowed')

  while(count > 1) {
    count = (count % 2 === 0) ? count / 2 : count * 3 + 1;
    step_count++;
  }

  return step_count;
}

File: solutions/python/collatz-conjecture/1/collatz_conjecture.py
def steps(number):
    if number <= 0:
        raise ValueError("Only positive integers are allowed")
    
    step_count = 0
    
    while number > 1:
        number = number // 2 if number % 2 == 0 else number * 3 + 1
        step_count += 1
    
    return step_count
